### PR TITLE
Use document noun

### PR DESCRIPTION
**Pull request by dragon-dxw**
Update the public UI to use document noun to identify press summaries etc.

### CODE REVIEW

**Comment by dragon-dxw** [comment]
Whilst the tests pass, smoketests have a failure due to custom-api-client quirks. Fixing...

File: judgments/tests/test_detail.py [hunk 1]
```diff
@@ -492,8 +512,8 @@ def test_document_headings_when_press_summary(
         self, mock_get_document_by_uri, mock_get_pdf_size
     ):
         """
-        GIVEN a press summary and related judgment
-        WHEN a request is made with the press summary URI
+        WHEN a request is made with to a document detail page
```
Comment (anthonyhashemi) [inline judgments/tests/test_detail.py:515]: could these 2 lines be swapped to be consistent with the Given-When-Then syntax?

File: judgments/utils.py [hunk 2]
```diff
@@ -212,4 +212,4 @@ def get_document_by_uri(document_uri: str) -> Judgment:
     )
 
     # raises a JudgmentNotFoundError if it doesn't exist
```
Comment (anthonyhashemi) [inline judgments/utils.py:214]: It now raises `DocumentNotFoundError`.

```suggestion
    # raises a DocumentNotFoundError if it doesn't exist
```

File: judgments/views/detail.py [hunk 3]
```diff
@@ -79,23 +84,26 @@ def detail(request, document_uri):
     context = {}
 
     press_summary_suffix = "/press-summary/1"
-    if document_uri.endswith(press_summary_suffix):
-        context["document_type"] = "press_summary"
+    context["document_noun"] = document.document_noun
+    if document.best_human_identifier is None:
+        raise NoNeutralCitationError(document.uri)
```
Comment (anthonyhashemi) [inline judgments/views/detail.py:89]: I think we should never reach the scenario of NoNeutralCitationError in a published document but this will make it easier for us to debug any edge cases that slip through for whatever reason.

Could we add tests for when we raise this then:

i.e for line 89 and 106.



even though we should not reach this scenario can we add a test for this scenario

File: judgments/views/detail.py [hunk 4]
```diff
@@ -79,23 +84,26 @@ def detail(request, document_uri):
...
         context["linked_document_uri"] = ""
-
-    if context["document_type"] == "press_summary" and linked_document:
-        context["judgment_ncn"] = linked_document.neutral_citation
+    else:
+        if context["document_noun"] == "press summary" and linked_document:
+            if linked_document.best_human_identifier:
+                context["judgment_ncn"] = linked_document.best_human_identifier
+            else:
+                raise NoNeutralCitationError(linked_document.uri)
```
Comment (anthonyhashemi) [inline judgments/views/detail.py:106]: as mentioned in other comment, could we test this

File: judgments/utils.py [hunk 5]
```diff
@@ -212,4 +212,4 @@ def get_document_by_uri(document_uri: str) -> Judgment:
...
-    return Judgment(document_uri, api_client)
+    return api_client.get_document_by_uri(document_uri)
```
Comment (anthonyhashemi) [inline judgments/utils.py:215]: looks like we don't currently test
`get_document_by_uri` and we mock it out in any test of functions that call it.

Can we add the following tests for it:


```
import unittest
from judgments.utils import get_document_by_uri

class TestGetDocumentByUriOrRaise(unittest.TestCase):
    @mock.patch('judgments.utils.MarklogicApiClient')
    def test_get_existing_document(self, mock_api_client):

        document = Mock()
        mock_api_client.return_value.get_document_by_uri.return_value = document

        result = get_document_by_uri_or_raise('sample_uri')

        self.assertEqual(result, document)

    @mock.patch('judgments.utils.MarklogicApiClient')
    def test_get_nonexistent_document(self, mock_api_client):
        mock_api_client.return_value.get_document_by_uri.side_effect = DocumentNotFoundError

        with self.assertRaises(DocumentNotFoundError):
            get_document_by_uri_or_raise('nonexistent_uri')
```

File: judgments/views/detail.py [hunk 6]
```diff
@@ -79,23 +84,26 @@ def detail(request, document_uri):
...
 
-    linked_document = None
     try:
         linked_document = get_published_document_by_uri(context["linked_document_uri"])
     except Http404:
         context["linked_document_uri"] = ""
-
-    if context["document_type"] == "press_summary" and linked_document:
-        context["judgment_ncn"] = linked_document.neutral_citation
+    else:
```
Comment (anthonyhashemi) [inline judgments/views/detail.py:101]: I think we can skip the else to reduce the levels of nesting

the line `if context["document_noun"] == "press summary" and linked_document:` 

checks for the linked_document being non Falsey which can only be the case if the try block succeeded.

File: judgments/views/detail.py [hunk 7]
```diff
@@ -79,23 +84,26 @@ def detail(request, document_uri):
...
+        if context["document_noun"] == "press summary" and linked_document:
+            if linked_document.best_human_identifier:
```
Comment (anthonyhashemi) [inline judgments/views/detail.py:103]: On lone 88 we check for `if document.best_human_identifier is None:`

can we make these 2 checks consistent?

Comment (anthonyhashemi) [reply judgments/views/detail.py:103]: I probably wouldn't go as far as verifying the ncn is valid, but that does make me think again that we have validation in EUI to block publishing if ncn is invalid in any way, so maybe we could assume that best_human_identifier is valid for judgments and press summaries in the PUI and so NoNeutralCitationError is unnecessary?

**Review by jacksonj04 — COMMENTED** [review]
Drive-by comment so I know I've said it somewhere, rather than it should be addressed in this PR: all the places we're explicitly checking if the noun is "press summary" should probably be regarded as smelly in the mid-term, since this isn't extensible. The fix will probably vary case-by-case though, maybe type-specific templates or pushing logic to a different layer.

File: judgments/tests/test_detail.py [hunk 8]
```diff
@@ -601,3 +633,17 @@ def test_html_title_when_judgment(
         response = self.client.get("/eat/2023/1")
         html_title = "<title>Judgment A - Find case law</title>"
         assert_contains_html(response, html_title)
+
+
+class TestNoNeutralCitation(TestCase):
```
Comment (anthonyhashemi) [inline judgments/tests/test_detail.py:638]: I was about to say that this just covers line 89 in views/detail.py and line 106 is still untested.

However, we can actually just remove that whole block.

i.e remove  lines 103-106

```
    if context["document_noun"] == "press summary" and linked_document:
        if linked_document.best_human_identifier is None:
            raise NoNeutralCitationError(linked_document.uri)
        context["judgment_ncn"] = linked_document.best_human_identifier
```

since line 89 since we now use best_human_identifier

**Review by anthonyhashemi — CHANGES_REQUESTED** [review]
All looks good now apart from we can remove the now redundant lines 103 to 106 in `detail.py`

File: judgments/views/detail.py [hunk 9]
```diff
@@ -79,24 +84,22 @@ def detail(request, document_uri):
...
-        context["judgment_ncn"] = document.neutral_citation
+        linked_doc_url = document_uri + press_summary_suffix
 
-    linked_document = None
     try:
-        linked_document = get_published_document_by_uri(context["linked_document_uri"])
+        linked_document = get_published_document_by_uri(linked_doc_url)
+        context["linked_document_uri"] = linked_document.uri
     except Http404:
+        linked_document = None
```
Comment (anthonyhashemi) [inline judgments/views/detail.py:100]: so now we don't need `linked_document` as a variable in the except case as it is now onlyu usef used for `linked_document_uri` so we can remove this line 100 `linked_document = None`

**Review by anthonyhashemi — APPROVED** [review]
